packages/backend-defaults/src/entrypoints/urlReader/lib/GerritUrlReader.ts

```diff
@@ -217,4 +236,84 @@ export class GerritUrlReader implements UrlReaderService {
       stripFirstDirectory: false,
     });
   }
+
+  private async searchFilesFromGitiles(
+    url: string,
+    options?: UrlReaderServiceReadTreeOptions,
+  ) {
+    const { branch, filePath, project } = parseGerritGitilesUrl(
+      this.integration.config,
+      url,
+    );
+
+    /** TODO: export something else from integrations */
+    const gitilesUrl = getGitilesAuthenticationUrl(this.integration.config);
+    const treeUrl = `${gitilesUrl}/${project}/+/refs/heads/${branch}/?format=JSON&recursive`;
+
+    const treeResponse = await fetch(treeUrl, {
+      ...getGerritRequestOptions(this.integration.config),
+      // TODO(freben): The signal cast is there because pre-3.x versions of
+      // node-fetch have a very slightly deviating AbortSignal type signature.
+      // The difference does not affect us in practice however. The cast can
+      // be removed after we support ESM for CLI dependencies and migrate to
+      // version 3 of node-fetch.
+      // https://github.com/backstage/backstage/issues/8242
+      signal: options?.signal as any,
+    });
+    if (treeResponse.status === 404) {
+      throw new NotFoundError(`Not found: ${gitilesUrl}`);
+    }
+    if (!treeResponse.ok) {
+      throw new Error(
+        `${url} could not be read as ${treeUrl}, ${treeResponse.status} ${treeResponse.statusText}`,
+      );
+    }
```

Comment:
so that other errors aren't mapped to 500:

```suggestion
    if (!treeResponse.ok) {
      throw await ResponseError.fromResponse(treeResponse);
    }
```